service/controller/app/resource/chartoperator/resource.go

```diff
@@ -194,3 +208,33 @@ func (r Resource) uninstallChartOperator(ctx context.Context, cr v1alpha1.App) e
 
 	return nil
 }
+
+func (r Resource) deleteFinalizers(ctx context.Context, cr v1alpha1.App) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	chart, err := cc.Clients.K8s.G8sClient().ApplicationV1alpha1().Charts(r.chartNamespace).Get(ctx, cr.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// no-op
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(chart.GetFinalizers()) > 0 {
+		r.logger.Debugf(ctx, "deleting remaining finalizers on %#q", key.AppName(cr))
+
+		chart.Finalizers = nil
+
+		_, err := cc.Clients.K8s.G8sClient().ApplicationV1alpha1().Charts(r.chartNamespace).Update(ctx, chart, metav1.UpdateOptions{})
```

Comment:
At some points, the chart-operator pod had been terminated without cleaning its finalizers from chart-operator.